packages/@aws-cdk/aws-s3-notifications/lib/lambda.ts

```diff
@@ -11,19 +11,28 @@ export class LambdaDestination implements s3.IBucketNotificationDestination {
   }
 
   public bind(_scope: Construct, bucket: s3.IBucket): s3.BucketNotificationDestinationConfig {
-    const permissionId = `AllowBucketNotificationsFrom${bucket.node.uniqueId}`;
+    const permissionId = `AllowBucketNotificationsTo${this.fn.permissionsNode.uniqueId}`;
 
-    if (this.fn.permissionsNode.tryFindChild(permissionId) === undefined) {
+    if (!Construct.isConstruct(bucket)) {
```

Comment:
Should we handle a case where bucket is **not** a construct? What can it be then? I need this check to pass `bucket` as the scope of the permission.

Reply:
@eladb I'd love you take on this

Reply:
This is fine. It can never be something else, so throwing an error is a good approach.

Reply:
👍